### DISCUSSION TITLE
How to handle aqua-proxy updates?

### DISCUSSION THREAD

**Discussion by frezbo** [aquaproj/aqua#2267] [Deprecated (Question and Answer)] [answered]
### Free text

`aqua update-aqua` doesn't seem to update aqua-proxy. I usually end up deleting it and running `aqua install <something>` which triggers a re-download of latest aqua-proxy

**Comment by suzuki-shunsuke** [comment, marked as answer]
The version of aqua-proxy is hardcoded to aqua itself.

https://github.com/aquaproj/aqua/blob/0608c94e05a834b962d177e8f581ba9ba6bbd993/pkg/installpackage/proxy.go#L15

`aqua i` command switches aqua-proxy version if the current version is different from the hardcoded version.

**Reply by suzuki-shunsuke** [reply]
If `aqua exec` also switches aqua-proxy version, then you don't need to run `aqua i`.

But I also think `aqua exec` should be as fast as possible, and we should avoid to add the overhead to this command.
I'm not sure if we can allow this overhead.

aqua-proxy is stable and isn't updated so frequently. And the update is not so important.
We rarely add new features to aqua-proxy, and users don't need to worry about aqua-proxy's version.
So I think the current status is enough.

**Reply by frezbo** [reply]
makes sense, my expectation was that when aqua is updated and a newer version of proxy was available, running any command installed via aqua would have updated proxy, anyways i can do a dummy `aqua i` to update it. Thanks for clarifying.
